### PR TITLE
Added unit test to validate YML configuration file.

### PR DESCRIPTION
**Pull request by irinaepshteyn**

### CODE REVIEW

**Comment by irinaepshteyn** [comment]
@sanjeevchopra , @amiri-ge ,

This is a freelancing effort to add a unit test to validate format of uaa.yml file that uaa is being packaged with.   The test is capable of detecting any outdated property formats as well as violation of property constraints. 

We could run it as a stand-alone test during uaa deployment right after we inject uaa with the environment specific configuration file:

```
cp uaa-cf-release/config-$DEPLOYMENT_TYPE/uaa.yml uaa/src/main/resources/
gradle test -Dtest.single=UaaValidateConfigurationTest
```
If test fails we would fail deployment job.

If it sounds reasonable, could you please review and merge into our release branch.

Thanks,
Irina

File: server/src/main/java/org/cloudfoundry/identity/uaa/impl/config/UaaConfiguration.java [hunk 1]
```diff
@@ -292,12 +292,18 @@ public static void main(String[] args) throws Exception {
         if (args.length != 1) {
             throw new IllegalArgumentException("YAML file required");
         }
+        Set<ConstraintViolation<UaaConfiguration>> errors = validateConfiguration(args[0]);
+        System.out.println(errors);
+    }
+
+    public static Set<ConstraintViolation<UaaConfiguration>> validateConfiguration(String filename)
```
Comment (sanjeevchopra) [inline server/src/main/java/org/cloudfoundry/identity/uaa/impl/config/UaaConfiguration.java]: minor:  rename fileName to yamlFilename

**Comment by sanjeevchopra** [comment]
Please submit a PR against cloudfoundry/uaa:develop as well.  Once that PR is merged, yml_validation_test_3.6.0 branch can be removed.

**Comment by irinaepshteyn** [comment]
retest this please

**Pull request closed without merging**